addons/sale/models/sale_order_line.py

```diff
@@ -593,27 +593,35 @@ def _compute_price_reduce_taxinc(self):
-            # remove packaging if not match the product
-            if line.product_packaging_id.product_id != line.product_id:
+            if not line.product_id or not line.product_uom_qty or not line.product_uom:
                 line.product_packaging_id = False
-            # Find biggest suitable packaging
-            if line.product_id and line.product_uom_qty and line.product_uom:
-                line.product_packaging_id = line.product_id.packaging_ids.filtered(
-                    'sales')._find_suitable_product_packaging(line.product_uom_qty, line.product_uom)
-
-    @api.depends('product_packaging_id', 'product_uom', 'product_uom_qty')
-    def _compute_product_packaging_qty(self):
-        for line in self:
-            if not line.product_packaging_id:
-                line.product_packaging_qty = False
             else:
-                packaging_uom = line.product_packaging_id.product_uom_id
-                packaging_uom_qty = line.product_uom._compute_quantity(line.product_uom_qty, packaging_uom)
-                line.product_packaging_qty = float_round(
-                    packaging_uom_qty / line.product_packaging_id.qty,
-                    precision_rounding=packaging_uom.rounding)
+                # remove packaging if not match the product
+                if line.product_packaging_id.product_id != line.product_id:
+                    line.product_packaging_id = False
+                # remove packaging if product_uom_qty is not a multiple of packaging_uom_qty
+                if line.product_packaging_id:
+                    packaging_uom = line.product_packaging_id.product_uom_id
+                    packaging_uom_qty = line.product_uom._compute_quantity(line.product_uom_qty, packaging_uom)
+                    product_packaging_qty = packaging_uom_qty / line.product_packaging_id.qty
+                    product_packaging_qty_integer = float_round(product_packaging_qty, precision_rounding=1.0)
+                    if not float_compare(product_packaging_qty,
+                                         product_packaging_qty_integer,
+                                         precision_rounding=line.product_uom.rounding) == 0:
+                        line.product_packaging_id = False
+                # if now no packaging, try find a suitable one
+                if not line.product_packaging_id:
+                    line.product_packaging_id = line.product_id.packaging_ids._find_suitable_product_packaging(line.product_uom_qty, line.product_uom)
+
+                # compute product_packaging_qty
+                if line.product_packaging_id:
+                    packaging_uom = line.product_packaging_id.product_uom_id
+                    packaging_uom_qty = line.product_uom._compute_quantity(line.product_uom_qty, packaging_uom)
+                    line.product_packaging_qty = packaging_uom_qty / line.product_packaging_id.qty
+                else:
+                    line.product_packaging_qty = 0
```

Comment:
You could create a private method on product packaging to avoid too much duplicate code between sale and purchase